### PR TITLE
Add amount difference to Sales and expected deposit account to Sale Accounts

### PR DESCRIPTION
**Pull request by leandrost**
[ch12202](https://app.clubhouse.io/nexaas/story/12202/1-recebíveis-adicionar-campos-faltantes-no-client)

**Changes:** 
- Add expected deposit account id to sale account
- Add to sale attributes the was missing(`installments`, `competency_month`, `amount_difference`) and removes attributes that api does not respond(`api_related`).

**Sale API response example:**
```json
[
    {
        "sale": {
            "custom_classifications": [],
            "id": 118092,
            "nominal_amount": "41.67",
            "interest_amount": null,
            "ticket_amount": null,
            "discount_amount": null,
            "total_amount": "41.67",
            "fixed_fee_amount": "0.21",
            "amount_difference": null,
            "net_amount": "41.33",
            "fee_percentage": "10.0",
            "fee_percentage_amount": "4.13",
            "receivable_amount": "32.2",
            "issuer": "American Express",
            "payment_method": "parcelled",
            "installment_number": 24,
            "installment_count": 24,
            "original_sale_id": 118068,
            "status": 1,
            "summary": false,
            "document": "",
            "document_emission_date": null,
            "provider_code": null,
            "description": "outro outro",
            "observation": "",
            "category_id": 530334,
            "classification_center_id": 17643,
            "person_id": 191988,
            "days_to_liquidation": 10,
            "sale_account_id": 12,
            "financial_account_id": 1915503,
            "liquidation_weekday": 1,
            "estimated_liquidated_at": "2020-05-11",
            "liquidated_at": null,
            "occurred_at": "2020-04-28",
            "confirmed_at": null,
            "created_at": "2018-05-28T11:58:52-03:00",
            "updated_at": "2018-06-13T21:54:50-03:00",
            "competency_month": "2018-05",
            "tax_charges": [
                {
                    "id": 30521,
                    "tax_id": 2,
                    "amount": "0.13",
                    "withholding": true
                },
                {
                    "id": 30522,
                    "tax_id": 3,
                    "amount": "0.21",
                    "withholding": true
                }
            ],
            "attachments": [],
            "installments": [],
            "links": [
                {
                    "rel": "previous",
                    "href": "https://sandbox.myfinance.com.br/entities/3703/sales/118069",
                    "method": "get"
                }
            ]
        }
    }
]
``

### CODE REVIEW

**Review by lmansur — DISMISSED** [review]

**Comment by leandrost** [comment]
[ch12202](https://app.clubhouse.io/nexaas/story/12202/1-recebíveis-adicionar-campos-faltantes-no-client)

File: lib/myfinance/entities/sale.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ class Sale < Base
       attribute :id
       attribute :installment_count
       attribute :installment_number
+      attribute :installments
```
Comment (lmansur) [inline lib/myfinance/entities/sale.rb]: I think this should be `attribute :installments, Array[Hash]`, since the API returns an Array of Hashes

Comment (leandrost) [reply lib/myfinance/entities/sale.rb]: Agree, all attributes should specify the type, I guess that I was just following the pattern and couldn't see that

**Review by vagnerzampieri — APPROVED** [review]

File: lib/myfinance/entities/sale.rb (hunk 1 above)
Comment (lmansur) [reply lib/myfinance/entities/sale.rb]: fixed in 474563e

**Review by lmansur — APPROVED** [review]

**Review by lmansur — APPROVED** [review]

**Comment by lmansur** [comment]
:shipit: released on [version 1.6.6](https://rubygems.org/gems/myfinance/versions/1.6.6)

**Pull request merged**